main.go

```diff
@@ -1,33 +1,35 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2020 Bonc.com.cn, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+
+/**
+ * @file    main_test.go.go
+ * @author  qihouliang(qihouliang@bonc.com.cn)
+ * @date    2023/4/11 22:26
+ * @brief
+ */
+
 package main
 
 import (
-	"context"
-	"os"
-	"os/signal"
-	"syscall"
-
+	"fmt"
 	"github.com/appleboy/CodeGPT/cmd"
+	"github.com/appleboy/CodeGPT/webhook"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
 )
 
-func withContextFunc(ctx context.Context, f func()) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
-		defer signal.Stop(c)
-
-		select {
-		case <-ctx.Done():
-		case <-c:
-			cancel()
-			f()
-		}
-	}()
-
-	return ctx
-}
-
 func main() {
-	ctx := withContextFunc(context.Background(), func() {})
-	cmd.Execute(ctx)
+	log.SetFlags(log.Lshortfile | log.LstdFlags)
+	cmd.InitConfig()
+	http.HandleFunc("/payload", webhook.HandleWebhook)
+	ip := viper.GetString("server.ip")
+	port := viper.GetString("server.port")
+	var address = fmt.Sprintf("%s:%s", ip, port)
+	fmt.Println("Listening for GitHub Webhooks on", address)
+	http.ListenAndServe(address, nil)
+
 }
```

Comment:
Overall, the code patch looks well-written and organized. However, there are a few improvements that could be made:

1. It is recommended to use a logger instead of printing to the console using fmt.Println(). This allows for more control over the output and makes it easier to debug if errors occur.

2. The withContextFunc() function seems unnecessary since it only wraps the context.WithCancel() and signal handling code. It could be simplified and removed to improve readability.

3. It is recommended to use HTTPS instead of HTTP for webhook communication to ensure data security and privacy.

4. The configuration settings should be validated and sanitized to prevent any security vulnerabilities such as injection attacks.

Overall, the code patch appears to be low-risk for bugs and security vulnerabilities.